cou/utils/juju_utils.py

```diff
@@ -149,6 +149,7 @@ class Unit:
     machine: Machine
     workload_version: str
     subordinates: List[SubordinateUnit] = field(default_factory=lambda: [], compare=False)
+    leader: bool = False
```

Comment:
I don't think this should have a default?  When retrieving the unit from the juju api, it will either be leader or not, and should be set accordingly.

Reply:
Removed